src/test/java/com/github/nylle/javafixture/InstanceFactoryTest.java

```diff
@@ -120,6 +124,17 @@ void fallbackToFactoryMethodWhenConstructorThrowsException() {
             assertThat(result.getValue()).isNotNull();
         }
 
+        @Test
+        @DisplayName("will fallback to factory method and pass exceptions on")
```

Comment:
This name is a bit misleading, because the class used in this test does not have a factory method.

If it had one and the factory method would succeed, then no exception would be thrown.
If it had one and the factory method would also throw an exception, the test is still green because the stack trace is the same (because we swallow the factory method exception [here](https://github.com/Nylle/JavaFixture/blob/736d4a2cccd44484397d8731e371dccca2d1675c/src/main/java/com/github/nylle/javafixture/InstanceFactory.java#L170)).

Reply:
I'm not quite sure I understand.

The test-class does not have a factory-method intentionally - because it wouldn't throw otherwise. Yet we still fall back to a factory-method, because everything else failed.
So the intention is to have all approaches fail (the factory-method approach being the final one) and display the entire stack to show what we tried.

Reply:
This test checks what happens if the constructor fails and the object has **no** factory method.

I would expect a different stack trace if a factory method was found but could not be successfully called because it also throws an exception.

Reply:
> This test checks what happens if the constructor fails and the object has no factory method.

Correct!

> I would expect a different stack trace if a factory method was found but could not be successfully called because it also throws an exception.

Me, too. But this is not what the test is supposed to test. Are you saying there is a test missing?

Reply:
Yes. We currently have no test for the unlikely case where a constructor method throws an exception and a factory method also throws one.
The more I think about it, the less I think this is ever going to be a problem.
We might just wait until someone files a bug report.

Reply:
I added the test.